contribs/gmf/src/services/authenticationservice.js

```diff
@@ -0,0 +1,247 @@
+ *     status: boolean
+ * }}
+ */
+gmf.AuthenticationDefaultResponse;
+
+
+/**
+ * @typedef {{
+ *     fullname: (string|null),
+ *     login: (string|null)
+ * }}
+ */
+gmf.User;
+
+
+gmfModule.value('gmfUser', {
+  'fullname': null,
+  'login': null
+});
+
+
+
+/**
+ * @constructor
+ * @extends {goog.events.EventTarget}
+ * @param {angular.$http} $http Angular http service.
+ * @param {string} authenticationBaseUrl URL to "authentication" web service.
+ * @param {gmf.User} gmfUser
+ * @ngInject
+ */
+gmf.Authentication = function($http, authenticationBaseUrl, gmfUser) {
+
+  goog.base(this);
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.$http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.baseUrl_ = authenticationBaseUrl;
+
+  /**
+   * @type {gmf.User}
+   * @private
+   */
+  this.user_ = gmfUser;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.changePasswordRouteSuffix_ = 'loginchange';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.isLoggedInRouteSuffix_ = 'isloggedin';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.loginRouteSuffix_ = 'login';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.resetPasswordRouteSuffix_ = 'loginresetpassword';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.logoutRouteSuffix_ = 'logout';
+
+  this.load_();
+};
+goog.inherits(gmf.Authentication, goog.events.EventTarget);
+
+
+/**
+ * Load the authentication service, which sends an asynch request to
+ * determine whether the user is currently connected or not.
+ * @private
+ */
+gmf.Authentication.prototype.load_ = function() {
+  var isLoggedInUrl = goog.uri.utils.appendPath(
+      this.baseUrl_, this.isLoggedInRouteSuffix_);
+  this.$http_.get(isLoggedInUrl, {
+    cache: false
+  }).then(
+      goog.bind(this.handleLogin_, this));
+};
+
+
+/**
+ * @param {string} newPwd
+ * @param {string} confPwd
+ * @return {angular.$q.Promise} Promise.
+ * @export
+ */
+gmf.Authentication.prototype.changePassword = function(newPwd, confPwd) {
+
+  var url = goog.uri.utils.appendPath(
+      this.baseUrl_, this.changePasswordRouteSuffix_);
+
+  // FIXME - use post, get is currently used for dev purpose
+  /*
+  return this.$http_.post(url, $.param({
+    newPassword: newPwd,
+    password: confirmNewPassword
+  }), {
+    headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+  });
+  */
+  return this.$http_.get(url);
+};
+
+
+/**
+ * @param {string} login
+ * @param {string} pwd
+ * @return {angular.$q.Promise} Promise.
+ * @export
+ */
+gmf.Authentication.prototype.login = function(login, pwd) {
+
+  var url = goog.uri.utils.appendPath(this.baseUrl_, this.loginRouteSuffix_);
+
+  // FIXME - use post, get is currently used for dev purpose
+  /*
+  return this.$http_.post(url, $.param({login: login, password: pwd}), {
+    headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+  }).then(goog.bind(
+  */
+  return this.$http_.get(url).then(
```

Comment:
use post instead.
It's ok tu use the `$.param({})` syntax as discussed previously.